src/platform/graphics/drm_formats.cpp

```diff
@@ -482,12 +484,16 @@ auto mg::DRMFormat::info() const -> std::optional<Info const>
     {
         return Info(info);
     }
-    /* TODO: This could fire quite frequently.
-     * Ideally we would have a rate-limited (or once-per-format) logger, but for now, just
-     * throw it in the debug stream.
-     */
-    mir::log_debug("Detailed info for format %s missing; please report so this can be added", name());
-    return {};
+
+    // Actually, we probably want `std::flat_set`, but no libstdc++ support yet.
+    static Synchronised<std::unordered_set<decltype(fourcc)>> unknown_formats_guard;
+    auto unknown_formats = unknown_formats_guard.lock();
+    if (!unknown_formats->contains(fourcc))
+    {
+        mir::log_warning("Detailed info for format %s missing; please report so this can be added", name());
+        unknown_formats->insert(fourcc);
```

Comment:
I know this is a micro optimisation but...
```suggestion
    if (unknown_formats->insert(fourcc).second)
    {
        mir::log_warning("Detailed info for format %s missing; please report so this can be added", name());
```

Also, maybe some guidance on where to report?

Reply:
I've put a link to GitHub issues in the message text. I *haven't* taken the micro-optimisation, because I think it makes the intent slightly less obvious and it is the micro-est of micro-optimisations.